client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -242,6 +181,63 @@ public boolean isFinished()
         return state.get() == State.FINISHED;
     }
 
+    @Override
+    public boolean validateCredentials(int timeout)
+            throws IOException
+    {
+        if (timeout <= 0) {
```

Comment:
When `timeout` is set to `0` in `Connection.isValid(timeout)` method, I will use `networkTimeout` instead of `0`. So it should always be positive. [setNetworkTimeout](https://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html#setNetworkTimeout)

Reply:
But that's not correct.  The API spec says that if the caller passes 0 as the timeout value, that means no timeout.

Reply:
Please refer to this: [Connection.setNetworkTimeout](https://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html#setNetworkTimeout(java.util.concurrent.Executor,%20int))

> Sets the maximum period a Connection or objects created from the Connection will wait for the database to reply to any one request.

There shouldn't be any truly unlimited operations. Ideally, `setNetworkTimeout(Executor, int)` should put executor in action and guard all operations, but this is not implemented in Trino code.